Automattic/newspack-plugin
# feat(ia): add theme and brand tab to settings page + theme selector component

### All Submissions:


### Changes proposed in this Pull Request:

- Added Theme and Brand tab on Settings page.
- Added theme selection with saving.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets.
2. Ensure site is setup i.e. `wp newspack setup` has run. This will guarantee that `newspack-theme` has been installed + activated. If any other theme is selected a fatal will be thrown during API requests to `/wp-json/newspack/v1/wizard/newspack-setup-wizard/theme` due to this tightly coupled code:

https://github.com/Automattic/newspack-plugin/blob/674136ad9b5c753f1b37c4d21af322510778cafe/includes/wizards/class-setup-wizard.php#L328-L334

3. Navigate to _/wp-admin/admin.php?page=newspack-settings#/theme-and-brand_, confirm below is present:

<img width="1097" alt="Screenshot 2024-08-23 at 17 01 48" src="https://github.com/user-attachments/assets/c6b21578-93f9-482b-8825-ec718dbf809c">

4. Furthermore, confirm switching themes and saving work as expected.
5. Enter Setup mode, the easiest way is to set `isPartOfSetup` to false in:

https://github.com/Automattic/newspack-plugin/blob/0d08c85da6524bf1a8da300a0e62a6d4a8b5c5ab/src/wizards/newspack/views/settings/theme-and-brand/index.tsx#L23

6. Rebuild assets and refresh browser. Observe homepage patterns are visible and function correctly e.g. switching layouts, selection saves between refreshes and tab transitions.

<img width="1056" alt="Screenshot 2024-08-28 at 14 17 47" src="https://github.com/user-attachments/assets/02132b56-90e5-4b2d-8530-204eb69c5cb3">

### Other information:
